### DISCUSSION TITLE
Chromecast Support

### DISCUSSION THREAD

**Discussion by v5point0** [PaulWoitaschek/Voice#1317] [Ideas]
Being able to stream to the TV for home consumption circumvents to a certain extent of not having a desktop counterpart.

**Comment by PaulWoitaschek** [comment]
What do you mean by that? Just stream the sound to the TV? So each music player has to implement a direct link to Chromecast to stream the music?

**Comment by v5point0** [comment]
PocketCasts > https://play.google.com/store/apps/details?id=au.com.shiftyjelly.pocketcasts and Plex > https://play.google.com/store/apps/details?id=com.plexapp.android are able to cast their local downloaded content to the TV via Chromecast. I was suggesting that Material Audiobook Player be able to do this too.

By being able to "cast" to your TV, we do not need necessarily need a desktop app at this stage though it would be welcomed in the future via Chrome OS or hopefully the Chrome browser eventually.

For Android users, the lack of an Audiobook Player that syncs progress/position across devices including mobile, desktop and tv platforms is a void that was left by iTunes especially if one was previously a iOS user. I believe Material Audiobook Player's integration with Chromecast (TV) and Chrome OS/Chrome Browser (Desktop) would be able to fill that void.

**Comment by v5point0** [comment]
With lollipop, I notice the Android OS cast feature carries on despite the screen being locked. This is great news, which means a native implementation of MAP casting feature would not be required.

**Comment by PaulWoitaschek** [comment]
Work on this has begun https://github.com/Ph1b/MaterialAudiobookPlayer/tree/chromecast

**Comment by relan** [comment]
Use of com.google.android.gms:play-services-cast effectively makes this project proprietary software. Can this feature be disabled at compile time?

**Comment by DePingus** [comment]
I'm with relan on this. If com.gogole.android.gms:play-services-cast can't be removed on compile then its a safe bet that this will get (sadly) dropped from F-Droid.

**Comment by Dunofrey** [comment]
Any update on when /if chromecast support is coming. Thanks Duncan

**Comment by PaulWoitaschek** [comment]
Hey Duncan, I added Android Audio Support yesterday. And I _think_ that that means that a good part of work for Chromecast is done already. So you shouldn't wait for ever ;-)

**Comment by philipobenito** [comment]
+1

**Comment by PaulWoitaschek** [comment]
Please use the GitHub +1 reactions. I'll getting enough emails so far ;-)

**Comment by philipobenito** [comment]
Sorry, on my phone, I'm finished now anyway :-) may have a look at putting in a couple of PR when I get some spare time.

**Comment by PaulWoitaschek** [comment]
That would be great. I have a license key so tell me when you need it.

**Comment by sebastianknief** [comment]
Hello there,
I just got a Chromecast Audio and was under the impression from the above that Material Audiobook Player supported this? I cannot see a "cast" button anywhere...am I missing something?

Thanks, and: Merry Christmas!

Sebastian

**Comment by PaulWoitaschek** [comment]
Hey Sebastian, no - that's not there yet. Merry Christmas!

**Comment by PaulWoitaschek** [comment]
I started integrating it [here](https://github.com/PaulWoitaschek/MaterialAudiobookPlayer/tree/feature/chromecast).
Next part is passing the actual audio to the chromecast.

**Comment by grepper** [comment]
I've just been using the built in 'cast to screen' feature of android for a several android versions now.  This is casting to my chromecast audio device, not sure what it would do with the AV version of chromecast. It pops up a notification that 'casting has not been optimized for audio', but it works just fine for me for audiobooks. But an audio adaptation for Material Player would be great! Love the app BTW.

Robert

**Comment by versavius** [comment]
Chromecast is a standard part of android, so perhaps its not so difficult to add a cast button in the app just like Plex and Rock Radio does to cast to a Google Chromecast device.

Would be a neat feature as i wont need a separate speaker to play my audiobooks, just straight to TV !

**Comment by NiveouSs** [comment]
Really missing this feature. Currently using a separate app when I need to cast to my Chromecast Audio, which is awkward.

**Comment by Nikify** [comment]
Any clues as to if or when it would be possible to implement the cast feature? I would love to listen to audiobooks on the Google Home speaker that I have situated on might night table.

**Comment by Keavon** [comment]
Google Home added native audiobook playback for titles purchased through Google Play, but I am hoping to listen to my DRM-free audiobooks through the Home speaker with native Cast abilities and avoiding the (sometimes unstable) Cast Screen feature.

**Comment by Dunofrey** [comment]
Hey Keavon, unfortunately Google doesn't offer a streaming API for Assistant/ Home devices. That's why services like PocketCasts haven't developed an Assistant Action. Also you would need a cloud server for the content, or at least you would have to host your content on a supported cloud storage service.

The conspiracy theories in my thinks this may be an attempt to make sure content services can't go around Google's payment systems / compete with Google.

Imagine if Audiable developed a Google Home Action. That will kill Google Play AudioBooks almost entirely.

**Comment by dmizz** [comment]
any updates on this? looks like it's been in the works for years

**Comment by evra** [comment]
May be #924 can help. Google drive can provide direct links which can be used by chromecast directly. Not sure about security though.

**Comment by mikec85** [comment]
any updates on this? looks like it's been in the works for years

**Comment by hkay-dev** [comment]
Was this FR abandoned? No worries if it is, just not clear from reading through the discussion.

**Comment by versavius** [comment]
This is a feature alot of commercial apps have got so it would be nice to have it added, not a deal breaker in my opinion.
